motr/io_nw_xfer.c

```diff
@@ -1347,11 +1353,12 @@ static int nw_xfer_io_distribute(struct nw_xfer_request *xfer)
 			if (rc != 0)
 				goto err;
 
-			if (op_code == M0_OC_WRITE && do_cobs)
-				m0_bitmap_set(&units_spanned, unit, true);
-
 			ti->ti_ops->tio_seg_add(ti, &src, &tgt, r_ext.e_start,
 						m0_ext_length(&r_ext), iomap);
+			if (op_code == M0_OC_WRITE && do_cobs &&
+			    ti->ti_req_type == TI_READ_WRITE)
+				m0_bitmap_set(&units_spanned, unit, true);
+
```

Comment:
So, this was a bug, because ti->ti_req_type is set to TI_READ_WRITE in ti->ti_ops->tio_seg_add(), right?

So, I suggest to add the following assert:
```
M0_ASSERT(M0_IN(ti->ti_req_type, (TI_READ_WRITE, TI_COB_CREATE));
```
after the ti->ti_ops->tio_seg_add().

Reply:
Good suggestion, will add the assert here.

Reply:
TI_COB_CREATE is set in nw_xfer_io_distribute(), if target is not TI_READ_WRITE, so can't add a assert here.

Reply:
OK